configure.ac

```diff
@@ -1828,8 +1828,9 @@ esac
 if test "$Py_LTO" = 'true' ; then
   case $CC in
     *clang*)
-      dnl flag to disable lto during linking
-      LDFLAGS_NOLTO="-fno-lto"
+      dnl For clang, it seems our LTO policy for bootstrap and freezing needs
+      dnl to match the rest of the build.
+      LDFLAGS_NOLTO="-flto=${Py_LTO_POLICY}"
```

Comment:
```suggestion
      dnl Clang linker requires -flto in order to link objects with LTO information.
      dnl Thin LTO is faster and works for object files with full LTO information, too.
      LDFLAGS_NOLTO="-flto=thin"
```

Reply:
All clang does not support thin LTO.
Please take a look at my PR: https://github.com/python/cpython/pull/96766